### PR TITLE
[MM-10354] Add feature to remove team icon

### PR DESCRIPTION
**Pull request by saturninoabril**
#### Summary
Add feature to remove team icon

@asaadmahmood I'd like to request for your help on fixing CSS styles on images. The functionality is there but the UI is far from the design. Note that the server and redux changes are still for approval.

screen shot 2018-04-28 at 6 47 14 am

#### Ticket Link
Jira ticket: [MM-10354](https://mattermost.atlassian.net/browse/MM-10354)

#### Checklist
- [x] Ran `make check-style` to check for style errors (required for all pull requests)
- [x] Ran `make test` to ensure unit and component tests passed
- [x] Added or updated unit tests (required for all new features)
- [x] Has server changes (https://github.com/mattermost/mattermost-server/pull/8684)
- [x] Has redux changes (https://github.com/mattermost/mattermost-redux/pull/474)
- [x] Has UI changes
- [ ] Includes text changes and localization file ([.../i18n/en.json](https://github.com/mattermost/mattermost-webapp/blob/master/i18n/en.json)) updates
- [ ] Touches critical sections of the codebase (auth, posting, etc.)

### CODE REVIEW

File: components/setting_picture.jsx [hunk 1]
```diff
@@ -180,17 +200,31 @@ export default class SettingPicture extends Component {
             );
         }
 
+        let helpText;
+        if (imageContext === 'team') {
+            helpText = (
+                <FormattedHTMLMessage
+                    id={'setting_picture.help.team'}
+                    defaultMessage='Upload a team icon in BMP, JPG or PNG format.<br>Images with a solid background color are recommended.'
```
Comment (hmhealey) [inline components/setting_picture.jsx]: This is a design question for PMs, but couldn't we just fill in the background of transparent images with white like we do for profile pictures?

@esethna

Comment (esethna) [reply components/setting_picture.jsx]: Yes, that was part of this [ticket](https://mattermost.atlassian.net/browse/MM-10354)

**Comment by saturninoabril** [comment]
Server and redux changes are recently merged. I've updated redux's commit to take effect remove team icon functionality.

File: components/setting_picture.jsx [hunk 2]
```diff
@@ -143,6 +144,25 @@ export default class SettingPicture extends Component {
                     src={this.props.src}
                 />
             );
+
+            if (this.props.onRemove) {
+                img = (
+                    <div className={`${imageContext}-img__container`}>
+                        <img
+                            ref='image'
```
Comment (lieut-data) [inline components/setting_picture.jsx]: Do we actually use this `ref` anywhere?

Comment (saturninoabril) [reply components/setting_picture.jsx]: Just realized not, I removed it now. Thanks!

**Comment by esethna** [comment]
Thanks @saturninoabril @asaadmahmood!

**Comment by saturninoabril** [comment]
Thanks to you too @esethna for the reviews and also to @asaadmahmood for big help on styling!
PR should be good to merge now.

**Pull request merged**